src/lib.rs

```diff
@@ -87,11 +87,16 @@ fn pack_size(n: u64) -> usize {
     }
 }
 
+mod private {
+    /// Sealed stops crates other than byteorder from implementing any traits that use it.
+    pub trait Sealed{}
+    impl Sealed for super::LittleEndian {}
+    impl Sealed for super::BigEndian {}
+}
+
 /// ByteOrder describes types that can serialize integers as bytes.
-///
-/// Note that `Self` does not appear anywhere in this trait's definition!
-/// Therefore, in order to use it, you'll need to use syntax like
-/// `T::read_u16(&[0, 1])` where `T` implements `ByteOrder`.
```

Comment:
oops, not sure how I managed that; fixed. Vim-fat-fingers and not reviewing my own commit probably.